### DISCUSSION TITLE
How can I automatically format my .qmd (Quarto Markdown) files in VS Code?

### DISCUSSION THREAD

**Discussion by cigit-zgy** [quarto-dev/quarto-cli#10948] [Q&A]
### Description

How can I automatically format my .qmd (Quarto Markdown) files in VS Code?

**Comment by danmackinlay** [comment]
Depends what you mean by _format_, but I would start with the [quarto VS Code extension](https://marketplace.visualstudio.com/items?itemName=quarto.quarto)

**Reply by mcanouil** [reply]
Any Markdown formatter should work and specific language formatter as well for code cells.
In the end, those tools are accessible in various way depending on your setup.

A simple online search should get you started.

**Reply by cigit-zgy** [reply]
Thanks for your kind response. I'd like to set up a automatic formatter for .qmd files.

**Reply by mcanouil** [reply]
Look for "format on save" option.

In my opinion, "format on save" is more prone to error than doing it explicitly.
I would personally discourage to do this, especially because no formatter is aware of all languages special formatting.
For instance, you use `styler` for R code cells, but Black (or other) for Python, etc.
